Code/Tools/AssetProcessor/native/utilities/BuilderManager.cpp

```diff
@@ -433,21 +83,23 @@ namespace AssetProcessor
         {
             AZStd::lock_guard<AZStd::mutex> lock(m_buildersMutex);
```

Comment:
since this guard is protecting the Builder List, should it go into the Builder List class?

Reply:
I went back and forth on that a few times.  I'd like to put it inside but then I'd have to encapsulate a lot more code in the class in order to be sure the lock is held long enough, which probably gets outside of its responsibilities.